### DISCUSSION TITLE
firmware update - now motors will not power on

### DISCUSSION THREAD

**Discussion by cswiger** [bitcraze/discussions#1481] [Q&A] [answered]
had crazyflie 2.0 for years, says 3/22/2016 on the box, seeed studio - and decided to use it for programming. On Ubuntu 22.04 installed cfclient and was able to connect ok using the USB dongle. Saw an option for FW update, which I would have to do for a flow deck on order anyway, and that completed, restarted twice.    Now when I try to fly everything seems to connect ok, but the throttle does nothing, the bird is inert at this point.   Tried resetting back to an older version trying the oldest in the list 2019.02.2 cf2 - test, no throttle.

**Comment by cswiger** [comment]
Luckily I have a parts donor with a board that works! Then got that to work with the cfclient and USB dongle from Ubuntu, but too scared to flash fw now :/       The scrogged board I tried a middle version like 2022.03 or something and that completely wiped it with two red and one blue I think, but no longer reachable.  Tried to enter bootload or firmware but have not the knack yet.

**Reply by knmcguire** [reply]
hi! Great that you are getting back into it again :D 

We also have a trouble shooting guide which also has instructions in case the crazyflie is bricked: https://www.bitcraze.io/support/troubleshooting/. Could you try that and see if it works to at least but it in boatloader mode?

**Reply by cswiger** [reply]
Thanks - what finally worked was:   I have to hold power button when connecting battery to the cf2.0 to get the bootloader, and using USB cable is much easier :)       Then both blue were flashing I think - whereas holding power btn with one hand and connecting battery with the other (challenging!)  only got a single flashing blue - first slow, then faster, then cfclient would never connect for update. 

this worked with USB power to cf2.0 and both blue lights flashing:

~/src/crazyflie-clients-python$ ./bin/cfloader flash fw/firmware-cf2-2024.2.zip 
Restart the Crazyflie you want to bootload in the next
 10 seconds ...
Found manifest version: 2
nRF51 has: sd-s110 and requires sd-s130 and upgrade provides sd-s130. Current bootloader version is [None] but upgrade provides [2024.2]
Should flash nRF soft device
Flashing 1 of 1 to nrf51 (bootloader+softdevice): 1023 bytes (2 pages) .1
Flashing 1 of 1 to nrf51 (bootloader+softdevice): 125951 bytes (124 pages) .1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1
Closing bootloader link and reconnecting to new bootloader (radio://0/0/2M/B1D71F45B3?safelink=0)
Reconnected to new bootloader
Flashing 1 of 2 to stm32 (fw): 271455 bytes (266 pages) ..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10..........10......6
Flashing 2 of 2 to nrf51 (fw): 46079 bytes (46 pages) .1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1.1
Skipping updating deck on coldboot

**Reply by cswiger** [reply]
With that I'm back to square one - previously the CF2.0 board worked fine with cfclient, I could use a joy stick and fly OK.

Then I first programmed fw just to test ability to do with - with a Flow Deck on order - and that is when the throttle stopped working (then several successive re-flash with various versions got it totally scrogged, needed the above recovery)    

I have the Flow Deck 2.0 mounted properly, fw updated successfully - and advancing the throttle results in the blue bar in cfclient moves, but no motors activate. 

The goal is to be able to control the CF2.0 from python on the computer, or something like droneblocks. 

So status is - one cr2.0 board has been recovered and flashed with the oldest 2019.02 fw in the list, and the flow deck attached, but motors are inop, connect, throttle up to 80%, nothing happens, even when using the radio dongle with a cell phone known to work with the good board.     motors have not worked since the first flash, so I'm afraid that will break the good working one trying to get the Flow Deck 2.0 working on it.

**Reply by cswiger** [reply]
Tried just moving the Flow Deck and everything over to the working motor chassis, and motors come on but there is no option for "Hover Assist" in cfclient, I'm guessing because it has not been flashed with the flow deck attached, which we are afraid to do as that is likely to make the morors not work anymore!

just guessing - these two were a prize for finishing some tech task (try our product and get a free mini-drone!)   so maybe they are older models being surplussed and need an older fw?   The box is marked CF 2.0  and not 2.1.

**Reply by knmcguire** [reply]
Hmmm... technically the crazyflie 2.0 should be supported with the latest firmware so this would be strange. We have a crazyflie 2.0 in our CI hardware lab but I'm not sure if anyone flew with it recently (it's pretty empty at the office now).

If you connect to the crazyflie and post the content of the console tab and the debug tab? I'll try to see if I have a crazyflie 2.0

* The [cfclients](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/)'s debug tab (found under 'Help'>'about') 
* Copy pasting the[ console tab's content](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/console_tab/)

**Reply by cswiger** [reply]
Ok, - console shows an issue with a gyro:

MPU6500: Self test gyro Y [FAIL]. low: -14.0, high: 14.0, measured: -34.25

Could very well be failed hardware that did not show up until attempting the flash fw. 


SYS: I am 0x303235363135510D003C005B and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_DRIVERS: Found 16 drivers
DECK_INFO: Found 0 deck memory.
DECK_CORE: 0 deck enumerated
MPU9250 I2C connection [OK].
AK8963 I2C connection [OK].
LPS25H I2C connection [OK].
ESTIMATOR: Using estimator 1
CONTROLLER: Using controller 1
EEPROM: I2C connection [OK].
MPU6500: Self test gyro Y [FAIL]. low: -14.0, high: 14.0, measured: -34.25
AK8963: Self test [OK].
SYS: Free heap: 16056 bytes


DEBUG:
Cfclient
Cfclient version: 2024.7.1
System: linux
Python: 3.10.12
Qt: 6.7.1
PyQt: 6.7.1

Interface status
radio: Crazyradio version 0.53
UsbCdc: No information available
udp: None
prrt: No information available
cpx: None

Input readers
linuxjsdev (1 devices connected)

Input devices
linuxjsdev: (0) Logitech Logitech Extreme 3D

Crazyflie
Connected: radio://0/80/250K
Firmware: a1d5aac6f3e6 (MODIFIED)

Decks found

Sensors found
BMP388: 0
HMC5883L: 1
MS5611: 1

Sensors tests
MPU6500: 1
HMC5883L: 1
MS5611: 1

**Reply by cswiger** [reply]
Reset all and try again, this time no fail in console.
On power up the motors test spin OK, all looks well on the CF itself. 
'Propeller test' in console does nothing. 

EEPROM: I2C connection [OK].
AK8963: Self test [OK].
SYS: Free heap: 16056 bytes

I can *cause* the above gyro error just by powering on holding CF at an angle:
EEPROM: I2C connection [OK].
MPU6500: Self test acc Y [FAIL]. low: -14.0, high: 14.0, measured: -18.19
AK8963: Self test [OK].


Found task dump - all other console buttons are inactive.
SYSLOAD: Task dump
SYSLOAD: Load	Stack left	Name
SYSLOAD: 1.33 	295 	Tmr Svc
SYSLOAD: 95.29 	127 	IDLE
SYSLOAD: 0.0 	73 	PWRMGNT
SYSLOAD: 0.59 	211 	CRTP-RX
SYSLOAD: 0.47 	353 	STABILIZE
SYSLOAD: 0.0 	91 	CMDHL
SYSLOAD: 0.0 	175 	SENSORS
SYSLOAD: 0.0 	243 	MEM
SYSLOAD: 0.0 	111 	LOG
SYSLOAD: 0.0 	95 	PARAM
SYSLOAD: 0.79 	161 	SYSTEM
SYSLOAD: 0.02 	65 	CRTP-TX
SYSLOAD: 1.47 	47 	SYSLINK
SYSLOAD: 0.0 	113 	USBLINK

**Reply by knmcguire** [reply]
Yes, holding it at an angle and powering up will prevent the crazyflie from properly initializing the imu, so this is something to be expected.

Could you send the full output of the console tab? There might be other things in the initialization that you might have missed. Also, it might be better to just flash the latest firmware. This way, once I get a chance to try it out on a crazyflie 2.0, we know that it at least is on the same firmware.

**Reply by cswiger** [reply]
Sure - Here is the console tab out for both - This is the entire console of the *working* cf2.0 that was never flashed here:

h!                                                            <-- tried many times and could not get the output before this
CFGBLK: v1, verification [OK]
IMU: MPU9250 I2C connection [OK].
IMU: AK8963 I2C connection [OK].
IMU: LPS25H I2C connection [OK].
DECK_DRIVERS: Found 6 drivers
DECK_INFO: Found 0 deck memory.
DECK_CORE: 0 deck enumerated
EEPROM: I2C connection [OK].
AK8963: Self test [OK].
LPS25H: Self test [OK].
SYS: Free heap: 5264 bytes                   <-- the one that flys 


And here is the console out for the one that was flashed with the latest in the list 2024.2, and will not spin the motors:

SYS: Crazyflie 2.0 is up and running!
SYS: Build 0:a1d5aac6f3e6 (2019.02) MODIFIED
SYS: I am 0x303235363135510D003C005B and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_DRIVERS: Found 16 drivers
DECK_INFO: Found 0 deck memory.
DECK_CORE: 0 deck enumerated
MPU9250 I2C connection [OK].
AK8963 I2C connection [OK].
LPS25H I2C connection [OK].
ESTIMATOR: Using estimator 1
CONTROLLER: Using controller 1
EEPROM: I2C connection [OK].
AK8963: Self test [OK].
SYS: Free heap: 16056 bytes                <-- this one was flashed with 2024.02 and flies no more.

**Reply by cswiger** [reply]
THIS is full console for the working one!  scan, power off cf20, click 'connect', THEN power on the cf20 gets the full output:

SYS: ----------------------------
SYS: Crazyflie 2.0 is up and running!
SYS: Build NA:NA (2016.02) CLEAN
SYS: I am 0x3032353631355110320057 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
IMU: MPU9250 I2C connection [OK].
IMU: AK8963 I2C connection [OK].
IMU: LPS25H I2C connection [OK].
DECK_DRIVERS: Found 6 drivers
DECK_INFO: Found 0 deck memory.
DECK_CORE: 0 deck enumerated
EEPROM: I2C connection [OK].
AK8963: Self test [OK].
LPS25H: Self test [OK].
SYS: Free heap: 5264 bytes      <-- console power on output from working board.

**Reply by cswiger** [reply]
Found this old copy of 2016.02   https://git.initrobots.ca/dastonge/crazyflie-firmware.git
did the git clone, following instructions and chasing down errors :)

**Reply by cswiger** [reply]
Amazingly I did get the fw built, stock .config, flashed, and still no throttle from either cfclient or an android phone. 
Suspecting hw fail at this point! Here's the console out for the recent 2016.02 fw build:

SYS: ----------------------------
SYS: Crazyflie 2.0 is up and running!
SYS: Build 653:c2700513fadb (2022.05 +653) CLEAN
SYS: I am 0x303235363135510D003C005B and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 0 deck(s) found
MPU9250 I2C connection [OK].
AK8963 I2C connection [OK].
LPS25H I2C connection [OK].
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) contMTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes

**Reply by cswiger** [reply]
Sorry I messed up and did not set the github to the right version - did git checkout 2016.02, build it (needs python2 :)   and NOW THE MOTORS work!   The dog hates this thing :)       I did purchase another one from ebay just to get the flow deck working - will step forward thru the releases to find where it breaks.

**Reply by cswiger** [reply]
Tried other versions but 2016.02 is the only one that worked - others, if no issues building, had no motor  - but 2016.02 does not appear to support the flow deck 2, at least no option for 'hover' in cfclient.    SO these two prize giveaways must be factory rejects or an oddball rev nobody knew what to do with that ended up on the gray market - who knows :)

**Reply by knmcguire** [reply]
Well.. it makes sense because the flowdeck v2 came out in 2017, so the firmware and client software wouldn't support it yet in 2016. Here is a blogpost when it came out: https://www.bitcraze.io/2017/07/crazyflie-2-0-flow-deck/

I'm currently not at the office but once I am next week I'll try it out on a crazyflie 2.0 with the V2 flowdeck. I did use the combo myself in 2018 and that was working at the time for my phd research, so I'm sure we are just missing something.

**Reply by cswiger** [reply, marked as answer]
Ok - a new CF2.0 arrives and is working great - came with 2016.09 and updated to 2022.05 to pickup the flow deck and everything works fine - even the artificial horizon in cfclient where it did not with the two previous cf20.    So something is definitely oddball about those two prize giveaways!     Mission accomplished and everything working as expected.

**Reply by knmcguire** [reply]
ah oke! good at least you'll be able to work with these but strange it didn't work for those two. It might also be that those 2 you had might have been clones? But that would be the first we would have seen from the Crazyflie 2.0s...
